app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,357 @@
+            }
+        }
+    }
+}
+
+@Composable
+private fun PlantImage(
+    scrollerPosition: ScrollerPosition,
+    imageUrl: String,
+    modifier: Modifier = Modifier
+) {
+    ParallaxEffect(scrollerPosition, modifier) { parallaxModifier ->
+        CoilImageWithCrossfade(
+            data = imageUrl,
+            contentScale = ContentScale.Crop,
+            modifier = parallaxModifier.fillMaxWidth().preferredHeight(278.dp)
+        )
+    }
+}
+
+@Composable
+private fun HeaderBarContent(onBackClicked: () -> Unit, onShareClicked: () -> Unit) {
+    val iconModifier = Modifier.sizeIn(maxWidth = 32.dp, maxHeight = 32.dp)
+        .drawBackground(color = Color.White, shape = CircleShape)
+
+    Row(Modifier.fillMaxSize().padding(top = StatusBarHeight + 12.dp), Arrangement.SpaceBetween) {
+        IconButton(
+            onClick = onBackClicked,
+            modifier = Modifier.padding(start = 12.dp).plus(iconModifier)
+        ) {
+            Icon(Icons.Filled.ArrowBack)
+        }
+        IconButton(
+            onClick = onShareClicked,
+            modifier = Modifier.padding(end = 12.dp).plus(iconModifier)
+        ) {
+            Icon(Icons.Filled.Share)
+        }
+    }
+}
+
+@Composable
+private fun PlantInformation(
+    name: String,
+    wateringInterval: Int,
+    description: String,
+    onNamePositioned: (LayoutCoordinates) -> Unit,
+    toolbarShown: Boolean
+) {
+    Box(modifier = Modifier.padding(24.dp)) {
+        Hide(toolbarShown) { hideModifier ->
+            Text(
+                text = name,
+                style = MaterialTheme.typography.h5,
+                modifier = Modifier.padding(start = 8.dp, end = 8.dp, bottom = 16.dp)
+                    .gravity(Alignment.CenterHorizontally).onPositioned {
+                        onNamePositioned(it)
+                    }.plus(hideModifier)
+            )
+        }
+        Text(
+            text = stringResource(id = R.string.watering_needs_prefix),
+            color = MaterialTheme.colors.primaryVariant,
+            style = MaterialTheme.typography.body1.copy(
+                fontWeight = FontWeight.Bold
+            ),
+            modifier = Modifier.padding(horizontal = 8.dp).gravity(Alignment.CenterHorizontally)
+        )
+        Text(
+            text = getQuantityString(R.plurals.watering_needs_suffix, wateringInterval),
+            color = MaterialTheme.colors.onBackground.copy(alpha = 0.6f),
+            modifier = Modifier.padding(start = 8.dp, end = 8.dp, bottom = 16.dp)
+                .gravity(Alignment.CenterHorizontally)
+        )
+        PlantDescription(description)
+    }
+}
+
+@Composable
+private fun ParallaxEffect(
+    scrollerPosition: ScrollerPosition,
+    modifier: Modifier = Modifier,
+    content: @Composable (Modifier) -> Unit
+) {
+    val offset = scrollerPosition.value.px / 2
+    val offsetDp = with(DensityAmbient.current) { offset.value.toDp() }
+    content(Modifier.padding(top = offsetDp).plus(modifier))
```

Comment:
Use `offset()` here. `padding` will cause a layout pass which is going to hurt perf.

Reply:
I do actually need a layout pass here as the rest of the content on the screen needs to update accordingly.

Reply:
Ewwww

Reply:
Couldn't you use `offsetPx` here to avoid the `Dp` roundtrip?